src/test/java/com/puppycrawl/tools/checkstyle/filters/SuppressWithNearbyTextFilterTest.java

```diff
@@ -532,6 +543,63 @@ public void testCachedFileAbsolutePathHasChangedEachRun() throws IOException {
                 .isNotEqualTo(cachedFileAbsolutePath2);
     }
 
+    /**
+     * Calls the filter on a real input file and verifies that the
+     * filter caches the file path only once. Ensures 'cachedFileAbsolutePath' remains
+     * unchanged after the first violation and that the file text is retrieved only once.
+     *
+     * @throws IOException if an error occurs while formatting the path to the input file.
+     */
+    @Test
+    public void testCacheFilePathCalledOnlyOnce() throws IOException {
+        final SuppressWithNearbyTextFilter targetFilter = new SuppressWithNearbyTextFilter();
+        final FileRemovalAfterFirstUsageFilter fileRemovalFilter =
+                new FileRemovalAfterFirstUsageFilter();
+
+        final String originalInputPath = getPath(
+            "InputSuppressWithNearbyTextFilterDefaultConfig.java");
+        final File tempInputFile = new File(temporaryFolder,
+            "InputSuppressWithNearbyTextFilterDefaultConfig.java");
+        Files.copy(new File(originalInputPath).toPath(),
+                tempInputFile.toPath(), StandardCopyOption.REPLACE_EXISTING);
+
+        assertWithMessage("Temporary input file should exist after copying.")
+                .that(tempInputFile.exists())
+                .isTrue();
+
+        final String cachedFileAbsolutePath = getCachedFileAbsolutePathAfterExecution(
+                targetFilter, tempInputFile.getPath());
+        assertWithMessage("Cached file path should not be null or empty.")
+                .that(cachedFileAbsolutePath)
+                .isNotEmpty();
+
+        final AuditEvent auditEvent1 = new AuditEvent(
+                tempInputFile.getPath(), tempInputFile.getPath(),
+                new Violation(1, null, null, null, null, Object.class, null));
+        targetFilter.accept(auditEvent1);
```

Comment:
one more time: test method should use Input files with config inside of it.
https://github.com/checkstyle/checkstyle/commit/2d5bfc110a7b81e76e5821b9e899e65a71b24a86#diff-0aa5d4351017b80ba94f82da4e00e2396e2db8a811918444878635c740400263R1401

test should use `verifyWithInlineConfigParser` method only.

location to register test module - https://github.com/checkstyle/checkstyle/commit/2d5bfc110a7b81e76e5821b9e899e65a71b24a86#diff-e7b107f6c9570fd3f500483f55b8bfb863cae9c96384f286d9a50be3ed4b725fR61

example of Input file https://github.com/checkstyle/checkstyle/commit/2d5bfc110a7b81e76e5821b9e899e65a71b24a86#diff-fa96fe0c65d2dd977a14acfbd8d38cebb470c9b1fff52d19e22801b2d833120dR4